### ISSUE TITLE
[Issue]: Error in Embedded Image Extractor

### ISSUE THREAD

**Issue by Brancliff** [jellyfin/jellyfin#8440]
### Please describe your bug

I'm trying to fill out my Jellyfin with YouTube videos, and I'm using the Embedded Image Extractor with the intent to show the embedded thumbnail of the YouTube video as the main image for each episode. But whenever I do this, I get an error with the image extraction, and all of the episodes in the folder get the same episode image. I've attached some logs at the bottom of this post.

The weird thing is, when I have this problem, I can use ffmpeg to extract the thumbnail images myself just fine. So I'm not sure what's causing trouble with Jellyfin using ffmpeg to extract the thumbnail images. 

### Jellyfin Version

10.8.4

### Environment

```markdown
- OS: Linux
- Virtualization: Docker
- Clients: Browser
- Browser: Vivaldi v5.0.2497.32
- FFmpeg Version: MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Found ffmpeg version "5.1"
- Plugins: YouTubeMetadata (though, I don't think this should be a factor - the thumbnail info is already embedded into the file)
- Storage: Local
```

### Jellyfin logs

```shell
[15:02:41] [ERR] [64] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: ffmpeg image extraction failed for file:"/media/YouTube/Vargskelethor/Dwarf Fortress Adventure Mode/Vargskelethor 2015-12-05 Dwarf Fortress (Adventure Mode) (Part 4) [q2hM5bm1U4Q].mp4"
[15:02:41] [ERR] [64] MediaBrowser.Providers.TV.EpisodeMetadataService: Error in Embedded Image Extractor
MediaBrowser.Common.FfmpegException: ffmpeg image extraction failed for file:"/media/YouTube/Vargskelethor/Dwarf Fortress Adventure Mode/Vargskelethor 2015-12-05 Dwarf Fortress (Adventure Mode) (Part 4) [q2hM5bm1U4Q].mp4"
   at MediaBrowser.MediaEncoding.Encoder.MediaEncoder.ExtractImageInternal(String inputPath, String container, MediaStream videoStream, Nullable`1 imageStreamIndex, Nullable`1 threedFormat, Nullable`1 offset, Boolean useIFrame, Nullable`1 targetFormat, CancellationToken cancellationToken)
   at MediaBrowser.MediaEncoding.Encoder.MediaEncoder.ExtractImage(String inputFile, String container, MediaStream videoStream, Nullable`1 imageStreamIndex, MediaSourceInfo mediaSource, Boolean isAudio, Nullable`1 threedFormat, Nullable`1 offset, Nullable`1 targetFormat, CancellationToken cancellationToken)
   at MediaBrowser.Providers.MediaInfo.EmbeddedImageProvider.GetEmbeddedImage(Video item, ImageType type, CancellationToken cancellationToken)
   at MediaBrowser.Providers.Manager.ItemImageProvider.RefreshFromProvider(BaseItem item, IDynamicImageProvider provider, ImageRefreshOptions refreshOptions, TypeOptions savedOptions, ICollection`1 downloadedImages, RefreshResult result, CancellationToken cancellationToken)
```

**Comment by Shadowghost** [comment]
Can you post the file or a short sample which is failing so the error can be reproduced?

**Comment by Brancliff** [comment]
Hey! Thanks for coming to the thread.

I've uploaded a video that I've had this problem with to Google Drive. It's 622MB, which I hope isn't much of a problem - but if it is, let me know, and I can look for a smaller video that had the issue. It's an investigation about a strange gamer on Team Fortress 2 and it's really interesting :>

https://drive.google.com/file/d/1Xcmv8B8lLQ2i5w46TRWBkKy1zbaXGBh0/view?usp=sharing

Anyways, the thumbnail already embedded in the videofile, which is what I'm trying to get to output to Jellyfin, looks like this:

![Elmaxo 2022-08-12 The Story of TF2s Strangest Player  FrQJIIbzoyQ](https://user-images.githubusercontent.com/22332774/215246919-f72dd586-731e-465d-8418-5c33cec98d51.jpg)

However, what I actually ended up getting instead was this:

![folder](https://user-images.githubusercontent.com/22332774/215247098-dd5665c1-6a9d-4d29-8dae-8a2fb0455656.jpg)

Under Manage Library > Image fetches, the top priority is to use Embedded Image Extractor.

Additionally, this is a part of what was in my Jellyfin's container logs:

[04:59:12] [WRN] [17] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from http://10.0.0.147:4004/System/ActivityLog/Entries?startIndex=0&limit=7&minDate=2023-01-27T05%3A00%3A06.383Z&hasUserId=true to 10.0.0.241 in 0:00:07.2467635 with Status Code 200
[04:59:17] [WRN] [50] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from http://10.0.0.147:4004/Libraries/AvailableOptions?LibraryContentType=movies&IsNewLibrary=false to 10.0.0.241 in 0:00:03.232033 with Status Code 200
[04:59:23] [WRN] [41] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from http://10.0.0.147:4004/Items/ff0cc7b970aaa46586eec6083410b77b/Images/Primary?fillHeight=300&fillWidth=200&quality=96&tag=fa9ce1f2ce50c76e3f46b131603068f6 to 10.0.0.241 in 0:00:35.2179599 with Status Code 200
[04:59:30] [ERR] [63] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: ffmpeg image extraction failed for file:"/media/Documentaries/The Story of TF2s Strangest Player/Elmaxo 2022-08-12 The Story of TF2s Strangest Player [FrQJIIbzoyQ].mp4"
[04:59:30] [ERR] [63] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: I-frame image extraction failed, will attempt standard way. Input: file:"/media/Documentaries/The Story of TF2s Strangest Player/Elmaxo 2022-08-12 The Story of TF2s Strangest Player [FrQJIIbzoyQ].mp4"
MediaBrowser.Common.FfmpegException: ffmpeg image extraction failed for file:"/media/Documentaries/The Story of TF2s Strangest Player/Elmaxo 2022-08-12 The Story of TF2s Strangest Player [FrQJIIbzoyQ].mp4"
   at MediaBrowser.MediaEncoding.Encoder.MediaEncoder.ExtractImageInternal(String inputPath, String container, MediaStream videoStream, Nullable`1 imageStreamIndex, Nullable`1 threedFormat, Nullable`1 offset, Boolean useIFrame, Nullable`1 targetFormat, CancellationToken cancellationToken)
   at MediaBrowser.MediaEncoding.Encoder.MediaEncoder.ExtractImage(String inputFile, String container, MediaStream videoStream, Nullable`1 imageStreamIndex, MediaSourceInfo mediaSource, Boolean isAudio, Nullable`1 threedFormat, Nullable`1 offset, Nullable`1 targetFormat, CancellationToken cancellationToken)
[05:00:30] [ERR] [10] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: ffmpeg image extraction failed for file:"/media/Documentaries/The Story of TF2s Strangest Player/Elmaxo 2022-08-12 The Story of TF2s Strangest Player [FrQJIIbzoyQ].mp4"
[05:00:30] [ERR] [10] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: I-frame image extraction failed, will attempt standard way. Input: file:"/media/Documentaries/The Story of TF2s Strangest Player/Elmaxo 2022-08-12 The Story of TF2s Strangest Player [FrQJIIbzoyQ].mp4"
MediaBrowser.Common.FfmpegException: ffmpeg image extraction failed for file:"/media/Documentaries/The Story of TF2s Strangest Player/Elmaxo 2022-08-12 The Story of TF2s Strangest Player [FrQJIIbzoyQ].mp4"
   at MediaBrowser.MediaEncoding.Encoder.MediaEncoder.ExtractImageInternal(String inputPath, String container, MediaStream videoStream, Nullable`1 imageStreamIndex, Nullable`1 threedFormat, Nullable`1 offset, Boolean useIFrame, Nullable`1 targetFormat, CancellationToken cancellationToken)
   at MediaBrowser.MediaEncoding.Encoder.MediaEncoder.ExtractImage(String inputFile, String container, MediaStream videoStream, Nullable`1 imageStreamIndex, MediaSourceInfo mediaSource, Boolean isAudio, Nullable`1 threedFormat, Nullable`1 offset, Nullable`1 targetFormat, CancellationToken cancellationToken)
[05:00:30] [ERR] [35] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: ffmpeg image extraction failed for file:"/media/Documentaries/The Story of TF2s Strangest Player/Elmaxo 2022-08-12 The Story of TF2s Strangest Player [FrQJIIbzoyQ].mp4"
[05:00:30] [ERR] [35] MediaBrowser.Providers.Movies.MovieMetadataService: Error in Embedded Image Extractor
MediaBrowser.Common.FfmpegException: ffmpeg image extraction failed for file:"/media/Documentaries/The Story of TF2s Strangest Player/Elmaxo 2022-08-12 The Story of TF2s Strangest Player [FrQJIIbzoyQ].mp4"
   at MediaBrowser.MediaEncoding.Encoder.MediaEncoder.ExtractImageInternal(String inputPath, String container, MediaStream videoStream, Nullable`1 imageStreamIndex, Nullable`1 threedFormat, Nullable`1 offset, Boolean useIFrame, Nullable`1 targetFormat, CancellationToken cancellationToken)
   at MediaBrowser.MediaEncoding.Encoder.MediaEncoder.ExtractImage(String inputFile, String container, MediaStream videoStream, Nullable`1 imageStreamIndex, MediaSourceInfo mediaSource, Boolean isAudio, Nullable`1 threedFormat, Nullable`1 offset, Nullable`1 targetFormat, CancellationToken cancellationToken)
   at MediaBrowser.Providers.MediaInfo.EmbeddedImageProvider.GetEmbeddedImage(Video item, ImageType type, CancellationToken cancellationToken)
   at MediaBrowser.Providers.Manager.ItemImageProvider.RefreshFromProvider(BaseItem item, IDynamicImageProvider provider, ImageRefreshOptions refreshOptions, TypeOptions savedOptions, ICollection`1 downloadedImages, RefreshResult result, CancellationToken cancellationToken)
[05:01:01] [WRN] [103] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from http://10.0.0.147:4004/Items/2db04e625547e69aea8bfe224143f988/Images/Chapter/1?maxWidth=400&tag=879d16f1b7b4808006290aa65fe51a3b&quality=90 to 10.0.0.241 in 0:00:00.5761561 with Status Code 200
[05:01:01] [WRN] [94] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from http://10.0.0.147:4004/Items/0bd068bf84bd1fab1f9f463962a834d5/Images/Backdrop?fillHeight=203&fillWidth=360&quality=96&tag=f0bf94b372df808a202fa65e494bdccc to 10.0.0.241 in 0:00:05.8530822 with Status Code 200

**Comment by gunslinger123** [comment]
Confirming the bug exists on 10.8.11, downgrading to 10.8.10 and refreshing metadata, replacing all metadata, and checking replace existing images resolves the issue in my youtube library

**Comment by Brancliff** [comment]
Someone moved this issue to 'insufficient data'.
What more data should we give out?

**Comment by evrial** [comment]
Some random anime lover who hasn't touched a single line of code.

**Comment by cvium** [comment]
It works for me in unstable _but_, the file you have uploaded does not follow the *required* naming convention: https://jellyfin.org/docs/general/server/media/movies#images

The name of the stream must contain one of these keywords
```
            "poster",
            "folder",
            "cover",
            "default",
            "movie",
            "show"

**Comment by evrial** [comment]
When we can expect to get patch in stable? The recent release isn't fixed.

**Comment by evrial** [comment]
That's unfortunate that trivial regression getting fixed wihout ETA.

**Comment by evrial** [comment]
Youtube stores all videos in webm container in which idiots from google didn't supported cover images. So the output container will be remuxed to mkv with embed from 112 line:
`                '-metadata:s:%d' % new_stream, 'filename=cover.%s' % thumbnail_ext])`

```
  Stream #0:2: Attachment: none
    Metadata:
      filename        : cover.webp
      mimetype        : image/webp
```

**Comment by TheWicklowWolf** [comment]
> Youtube stores all videos in webm container in which idiots from google didn't supported cover images. So the output container will be remuxed to mkv with embed from 112 line: ` '-metadata:s:%d' % new_stream, 'filename=cover.%s' % thumbnail_ext])`
> 
> ```
>   Stream #0:2: Attachment: none
>     Metadata:
>       filename        : cover.webp
>       mimetype        : image/webp
> ```

In my case the video thumbnail is converted to png prior to embedding, but it still doesn't work...

yt_dlp/ffmpeg output:
```
[ThumbnailsConvertor] Converting thumbnail "myVideo.webp" to png
[EmbedThumbnail] mutagen: Adding thumbnail to "myVideo.mp4"
```

**Comment by TheWicklowWolf** [comment]
I'm not sure why this issue was closed as it hasn't been resolved.

Also it appears that there is a relatively simple solution: https://github.com/jellyfin/jellyfin/issues/10637


> With #10173 the image types bmp, gif, png and webp are no longer checked if the media type is not audio.
> Only mjpeg is checked for videos.
> 
> So if a video can have other formats embedded than mjpeg the code of [ProbeResultNormalizer.cs](https://github.com/jellyfin/jellyfin/pull/10173/files#diff-2bad799e56914e76976398c1a62474393b3950cd6ad6cd5d9acda67005295032R764) should probably look something like that:
> ```csharp
>                 if (isAudio
>                     || string.Equals(stream.Codec, "bmp", StringComparison.OrdinalIgnoreCase)
>                     || string.Equals(stream.Codec, "gif", StringComparison.OrdinalIgnoreCase)
>                     || string.Equals(stream.Codec, "png", StringComparison.OrdinalIgnoreCase)
>                     || string.Equals(stream.Codec, "webp", StringComparison.OrdinalIgnoreCase)))
>                 {
>                     stream.Type = MediaStreamType.EmbeddedImage;
>                 }
>                 else if (string.Equals(stream.Codec, "mjpeg", StringComparison.OrdinalIgnoreCase))
>                 {
>                     // How to differentiate between video and embedded image?
>                     // The only difference I've seen thus far is presence of codec tag, also embedded images have high (unusual) framerates
>                     else if (!string.IsNullOrWhiteSpace(stream.CodecTag))
>                     {
>                         stream.Type = MediaStreamType.Video;
>                     }
>                     else
>                     {
>                         stream.Type = MediaStreamType.EmbeddedImage;
>                     }
>                 }
> ```
> 
> _Originally posted by @TheMelmacian in https://github.com/jellyfin/jellyfin/issues/10637#issuecomment-1833559548_
>             
>

**Comment by TheMelmacian** [comment]
This was only an assumption based on the changed code that assumedly introduced the bug.
I haven't found any time so far to test it, so I can't say if this really fixes the problem.
If it works correctly in the unstable branch (10.9.*) as cvium mentioned, than this is not the solution since the mentioned code is still the same in the current master branch.

I know it's frustrating if an issue is closed without a solution but with [#10637](https://github.com/jellyfin/jellyfin/issues/10637) there is a still open duplicate to track the bug.

**Comment by evrial** [comment]
Issue is opened more than 1 year so recent code changes are questionable. Seems nobody cares from core developers.

**Comment by TheWicklowWolf** [comment]
> This was only an assumption based on the changed code that assumedly introduced the bug. I haven't found any time so far to test it, so I can't say if this really fixes the problem. If it works correctly in the unstable branch (10.9.*) as cvium mentioned, than this is not the solution since the mentioned code is still the same in the current master branch.
> 
> I know it's frustrating if an issue is closed without a solution but with [#10637](https://github.com/jellyfin/jellyfin/issues/10637) there is a still open duplicate to track the bug.

I'll keep an eye on that so

> Issue is opened more than 1 year so recent code changes are questionable. Seems nobody cares from core developers.

Fair enough really, its definitely not a major priority but it would be nice to get it sorted.

**Comment by cvium** [comment]
> Issue is opened more than 1 year so recent code changes are questionable. Seems nobody cares from core developers.

Nothing in that line of thought makes any sense. I have already told you I'm unable to reproduce it in unstable. We do not let fixed issues stay open even if the fix isn't in stable yet.
